smart-contract/contracts/Main.sol

```diff
@@ -230,32 +230,125 @@ contract ZuriSchoolVoting {
+                _user_type = "student";
+            } else if (voters[voter_addresses[i]].user_type == Stakeholder.TEACHER){
+                _user_type = "teacher";
+            } else {
+                _user_type = "director";
+            }
+            _user_types[i] = _user_type;
+            _canVotes[i] = voters[voter_addresses[i]].canVote;
+            _hasVoteds[i] = hasVoted[_electionId][voter_addresses[i]];
+        }
+        return (_names, _user_types, _canVotes, _hasVoteds);
+    }
 
 
 
-    // @function used to view results of an election by taking in the election id. should require the results to have been computed todo @Kosi
+     /**
+     * @notice compile the results of an election, the election is automatically stopped when the result is compiled
+     * @dev compile election results
+     * @param _electionId the id of the election you want to compile the results for
+     */
+    function compileResults(uint256 _electionId) public isDirectorOrTeacher {
+        stopElection(_electionId);
+        elections[_electionId].computed = true;
+
+        Proposal memory _max = choices[_electionId][0];
+        for(uint256 i = 1; i < choices[_electionId].length; i++){
+          if(choices[_electionId][i].voteCount > _max.voteCount){
+              _max = choices[_electionId][i];
+          }
+        }
 
+        winners[_electionId] = _max;
+        emit BallotResultCompiled(_electionId, elections[_electionId].name, block.timestamp);
+    }
 
 
-    // @function used to customize the vote weight for the various stakeholders todo @Kosi
+    /**
+     * @notice view the results of a completed election
+     * @dev view results of an election
+     * @param _electionId the id of the election you want to view the results for
+     */
+    function viewResult(uint256 _electionId) public view returns(string memory electionName, string memory proposalName, uint256 voteCount){
+        require(elections[_electionId].computed == true, "results have not yet been compiled");
+        return (elections[_electionId].name, winners[_electionId].name, winners[_electionId].voteCount);
+    }
 
 
+    /**
+     * @notice manually set the weights for a particular voter type
+     * @dev adjust voter type vote weights
+     * @param stakeholder is the name of the voter type you want to set the weights for
+     * @param weight is the value you want to set the weight to
+     */
+    function setWeight(string memory stakeholder, uint256 weight) public isChairperson {
+        require(weight >= 0, "weights can not be less than 1");
+        if(stringsEquals(stakeholder,"student")){
+            weights[Stakeholder.STUDENT] = weight;
+        } else if(stringsEquals(stakeholder,"teacher")){
+            weights[Stakeholder.TEACHER] = weight;
+        } else if(stringsEquals(stakeholder,"director")){
+            weights[Stakeholder.DIRECTOR]  = weight;
+        }else {
+            require(false, "invalid stakeholder name entered");
```

Comment:
What do you think about using `revert` here ?

Reply:
I actually don't know how to use revert, but you can merge it and make the changes from your end then ill pull it on my fork